DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/EmailDelegatesController.cs

```diff
@@ -16,33 +17,68 @@
     [Route("TrackingSystem/Delegates/Email")]
     public class EmailDelegatesController : Controller
     {
+        private const string EmailDelegateFilterCookieName = "EmailDelegateFilter";
+        private readonly CentreCustomPromptHelper centreCustomPromptHelper;
+        private readonly IJobGroupsDataService jobGroupsDataService;
         private readonly IPasswordResetService passwordResetService;
         private readonly IUserService userService;
 
-        public EmailDelegatesController(IUserService userService, IPasswordResetService passwordResetService)
+        public EmailDelegatesController(
+            CentreCustomPromptHelper centreCustomPromptHelper,
+            IJobGroupsDataService jobGroupsDataService,
+            IPasswordResetService passwordResetService,
+            IUserService userService
+        )
         {
-            this.userService = userService;
+            this.centreCustomPromptHelper = centreCustomPromptHelper;
+            this.jobGroupsDataService = jobGroupsDataService;
             this.passwordResetService = passwordResetService;
+            this.userService = userService;
         }
 
         [HttpGet]
-        public IActionResult Index()
+        public IActionResult Index(
+            string? filterBy = null,
+            string? filterValue = null,
+            bool selectAll = false
+        )
         {
+            filterBy = FilteringHelper.GetFilterBy(filterBy, filterValue, Request, EmailDelegateFilterCookieName);
+            var jobGroups = jobGroupsDataService.GetJobGroupsAlphabetical();
+            var customPrompts = centreCustomPromptHelper.GetCustomPromptsForCentre(User.GetCentreId());
             var delegateUsers = GetDelegateUserCards();
-            var model = new EmailDelegatesViewModel(delegateUsers);
+
+            var model = new EmailDelegatesViewModel(
+                delegateUsers,
+                jobGroups,
+                customPrompts,
+                filterBy,
+                selectAll
+            );
+
+            Response.UpdateOrDeleteFilterCookie(EmailDelegateFilterCookieName, filterBy);
 
             return View(model);
         }
 
         [HttpPost]
-        public IActionResult Index(EmailDelegatesViewModel model)
+        public IActionResult Index(EmailDelegatesViewModel model, string? filterBy = null, string? filterValue = null)
         {
             var delegateUsers = GetDelegateUserCards();
 
             if (!ModelState.IsValid)
             {
-                model.SetDelegates(delegateUsers);
-                return View(model);
+                filterBy = FilteringHelper.GetFilterBy(filterBy, filterValue, Request, EmailDelegateFilterCookieName);

```

Comment:
We shouldn't really be overriding the query string param value - can we declare a new var for the new filterBy string?